Libs/MRML/Core/vtkMRMLClipNode.cxx

```diff
@@ -688,3 +688,68 @@ void vtkMRMLClipNode::SetYellowSliceClipState(int state)
 {
   this->SetSliceClipState("vtkMRMLSliceNodeYellow", state);
 }
+
+//----------------------------------------------------------------------------
+bool vtkMRMLClipNode::GetClippingPlanes(vtkPlaneCollection* planeCollection)
+{
+  return this->GetClippingPlanesFromFunction(this->ImplicitFunction, planeCollection);
+}
+
+//---------------------------------------------------------------------------
+bool vtkMRMLClipNode::GetClippingPlanesFromFunction(vtkImplicitFunction* function, vtkPlaneCollection* planeCollection, bool invert/*=false*/)
```

Comment:
Could this method have an optional `vtkMRMLMessageCollection*` argument to return details that can be displayed to the user? `vtkMRMLVolumeRenderingDisplayNode` could call this method to get the messages.

The advantage compared to the current implementation would be that it would be easier to keep GetClippingPlanesFromFunction and IsClippingShortcutValid consistent.

Reply:
What messages would this function return? I could add in messages such as "no planes found", and "fast method cannot clip with more than 6 planes" messages, however the function GetClippingPlanesFromFunction is generic and not specifically related to fast clipping.